dusk/grammar.py

```diff
@@ -538,7 +538,10 @@ def field_index(self, field: DuskField, vindex=None, hindex=None):
                     voffset,
                     vbase,
                 )
-            return make_unstructured_offset(True), voffset, vbase
+            if self.ctx.location.is_dense(field_dimension):
+                return make_unstructured_offset(True), voffset, vbase
+            else:
+                return make_unstructured_offset(False), voffset, vbase
```

Comment:
```suggestion
            # sparse fields don't support (dense location) offsets
            return make_unstructured_offset(False), voffset, vbase
```
The if body is dead code, since `self.ctx.location.is_dense(field_dimension)` is guaranteed to be `False` at this point.

Added comment in the hopes that it makes things _clearer_ (probably doesn't, but at least we tried).